conans/test/functional/generators/virtualenv_windows_bash_test.py

```diff
@@ -0,0 +1,128 @@
+import os
+import platform
+import subprocess
+import textwrap
+import unittest
+
+from conans.test.functional.generators.virtualenv_test import _load_env_file
+from conans.test.utils.test_files import temp_folder
+from conans.test.utils.tools import TestClient
+from conans.util.files import decode_text
+
+
+@unittest.skipIf(platform.system() != "Windows", "Only for Windows")
+class VirtualenvWindowsBashTestCase(unittest.TestCase):
+    """
+    We are running the full example inside Bash (generation of environment files and activate/deactivate), so we need
+    to use actual Conan recipes and run an example. We cannot use the same approach as in 'virtualenv_test.py', but
+    we should test the same cases
+    """
+
+    maxDiff = None
+    conanfile = textwrap.dedent("""
+        import os
+        from conans import ConanFile
+        
+        class Recipe(ConanFile):
+            def build(self):
+                with open("executable.exe", "w+") as f:
+                    f.write("echo EXECUTABLE IN PACKAGE!!")
+            
+            def package(self):
+                self.copy("*.exe", dst="bin")
+
+            def package_info(self):
+                # Basic variable
+                self.env_info.USER_VAR = r"some value with space and \ (backslash)"
+                self.env_info.ANOTHER = "data"
+                
+                # List variable
+                self.env_info.WHATEVER = ["list", "other"]
+                self.env_info.WHATEVER2.append("list")
+                
+                # List with spaces
+                self.env_info.CFLAGS = ["cflags1", "cflags2"]
+                
+                # Add something to the path
+                self.env_info.PATH.append(os.path.join(self.package_folder, "bin"))
+                
+    """)
+
+    @classmethod
+    def setUpClass(cls):
+        cache_folder = os.path.join(temp_folder(path_with_spaces=False), ".conan")
+        t = TestClient(cache_folder=cache_folder)
+        t.save({"conanfile.py": cls.conanfile})
+        t.run("create . name/version@")
+
+        cls.cache_folder = os.path.dirname(t.cache_folder)
+
+    def _run_environment(self):
+        stdout, _ = subprocess.Popen(["where", "conan"], stdout=subprocess.PIPE).communicate()
+        conan_path = decode_text(stdout).splitlines()[0]  # Get the first one (Windows returns all found)
+
+        # All the commands are listed in a sh file:
+        commands_file = os.path.join(self.test_folder, 'commands.sh')
+        with open(commands_file, 'w+') as f:
```

Comment:
Same here, too many ``f.write()`` you can probably have a clean template that you fill with the folders, and then ``save(commands_file, content)``.

This is a matter of taste I am fine with it as-is too if you want to leave it.